xkafka/consumer_options.go

```diff
@@ -0,0 +1,102 @@
+package xkafka
+
+import (
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+	"github.com/pkg/errors"
+)
+
+// ConsumerOption is an interface for consumer options.
+type ConsumerOption interface{ setConsumerConfig(*consumerConfig) }
+
+type consumerConfig struct {
+	brokers         []string
+	configMap       kafka.ConfigMap
+	errorHandler    ErrorHandler
+	shutdownTimeout time.Duration
+	consumerFn      consumerFunc
+	topics          []string
+	metadataTimeout time.Duration
+	pollTimeout     time.Duration
+	concurrency     int
+	manualCommit    bool
+}
+
+func newConsumerConfig(opts ...ConsumerOption) (*consumerConfig, error) {
+	cfg := &consumerConfig{
+		consumerFn:      defaultConsumerFunc,
+		topics:          []string{},
+		brokers:         []string{},
+		configMap:       kafka.ConfigMap{},
+		metadataTimeout: 10 * time.Second,
+		pollTimeout:     10 * time.Second,
+		shutdownTimeout: 1 * time.Second,
+		concurrency:     1,
+	}
+
+	for _, opt := range opts {
+		opt.setConsumerConfig(cfg)
+	}
+
+	if err := cfg.validate(); err != nil {
+		return nil, err
+	}
+
+	return cfg, nil
+}
+
+func (c *consumerConfig) validate() error {
+	if len(c.brokers) == 0 {
```

Comment:
I think this may spiral into a long list of `if`s, it would be better to add a `validate() error` method on `type ConsumerOption` itself, wdyt?

```go
type ConsumerOption interface{
	setConsumerConfig(*consumerConfig)
	validate() error
}
```

Reply:
This will validate the value of the option, but cannot check if the option was set

Reply:
Ah, yes, missed this part, LGTM then.